File: A3/Sem5/Mobile/ciordales/mobile-backend1/src/main/java/com/example/mobilebackend1/dto/TicketDto.java
package com.example.mobilebackend1.dto;

import com.example.mobilebackend1.model.Ticket;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TicketDto {
    private String uuid;
    private String name;
    private String description;
    private Integer complexity;
    private Boolean isDone;
    private Double latitude;
    private Double longitude;
    private List<String> imagesUrls;
    private String created;
    private String updated;

    public TicketDto(Ticket ticket) {
        this.uuid = ticket.getUuid();
        this.name = ticket.getName();
        this.description = ticket.getDescription();
        this.complexity = ticket.getComplexity();
        this.isDone = ticket.isDone();
        this.latitude = ticket.getLatitude();
        this.longitude = ticket.getLongitude();
        this.imagesUrls = ticket.getImagesUrls();
        this.created = ticket.getCreated().toString();
        this.updated = ticket.getUpdated().toString();
    }
}

File: A3/Sem5/Mobile/ciordales/old/MobileAndroid/app/src/main/java/com/example/mobileandroid/dto/TicketDto.kt
package com.example.mobileandroid.dto

import java.io.Serializable

data class TicketDto(
    val uuid: String = "",
    val name: String = "",
    val description: String = "",
    val complexity: Int = 0,
    val isDone: Boolean = false,
    val latitude: Double = 0.0,
    val longitude: Double = 0.0,
    val imagesUrls: List<String> = listOf(),
    val created: String = "",
    val updated: String = "") : Serializable